### ISSUE TITLE
cp copy to symlinked folder

### ISSUE THREAD

**Issue by sakulstra** [shelljs/shelljs#414]
Hello,
I'm not sure if there's already an issue open for this topic (there are a few similar but the other way around - copy from symlink)

```
ln('-s', 'apache-karaf-4.0.4', '/root/apache-karaf');
cp('-r', '/root/modify-karaf/4.0.4/*', '/root/apache-karaf');
```

and get 
```
shell.js: internal error
Error: ENOENT, no such file or directory '/root/modify-karaf/4.0.4/*'
    at Error (native)
    at Object.fs.statSync (fs.js:797:18)
    at /usr/lib/node_modules/shelljs/src/cp.js:148:21
    at Array.forEach (native)
    at _cp (/usr/lib/node_modules/shelljs/src/cp.js:144:13)
    at /usr/lib/node_modules/shelljs/src/common.js:240:23
    at installKaraf (/root/install.js:140:5)
    at /root/install.js:40:5
    at GetAddrInfoReqWrap.asyncCallback [as callback] (dns.js:81:16)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:99:10)

```
from the cp command ```cp -r /root/modify-karaf/4.0.4/* /root/apache-karaf``` when I execute the exact same code manually in terminal it works without an error.

Regards,
Lukas

edit:
I think it's not symlink realted... I get the same error when using
```cp('-r', '/root/systemFiles/JAVA_HOME/*', env['JAVA_HOME']);```

My guess would be, that there's some problem when the wildcarded second parameter only contains folders.

**Comment by nfischer** [comment]
@sakulstra Thanks for reporting this! I think this bug is a duplicate of #389. Here's what I did (and you'll see it uses the latest version of master).

```bash
$ ls -RAF # here's my directory structure
.:
node_modules/  root/

./node_modules:

./root:
apache-karaf-4.0.4/  modify-karaf/

./root/apache-karaf-4.0.4:

./root/modify-karaf:
4.0.4/

./root/modify-karaf/4.0.4:
first.txt  second.txt  third.txt

$ npm install github:shelljs/shelljs # install the latest build

$ n_shell # this will use the version we just installed
Warning: using shelljs found at /home/nate/scratch/414/node_modules/shelljs
shelljs $ ln('-s', 'apache-karaf-4.0.4', 'root/apache-karaf');
{ [String: '']
  stdout: '',
  ...
shelljs $ cp('-r', 'root/modify-karaf/4.0.4/*', 'root/apache-karaf');
{ [String: '']
  stdout: '',
  ...
shelljs $ exit();

$ ls -RAF root # we can see it was successful!
root:
apache-karaf@  apache-karaf-4.0.4/  modify-karaf/

root/apache-karaf-4.0.4:
first.txt  second.txt  third.txt

root/modify-karaf:
4.0.4/

root/modify-karaf/4.0.4:
first.txt  second.txt  third.txt
# woohoo!
```

If this is not fixed against the latest version of master (maybe I misunderstood your bug report), let me know. Otherwise, this fix will be released in v0.7, which should be out this week. I'll close this issue for now, but please reopen if this truly is not fixed.

**Comment by nfischer** [comment]
Whoops! Forgot to mention that [n\_shell](https://github.com/nfischer/n_shell) starts a node REPL with shelljs included by default. The warning message means that it's using the locally installed version, which we want in this example (if there's no warning message, then it probably won't work as expected!).

**Comment by sakulstra** [comment]
@nfischer: damn, sry didn't see this issue.
It's the same error as in the issue, so it's probably already fixed - to verify I have to wait till 0.7 is out as i don't wont to pollute our buildserver with alternative versions :)

Thx for your help!

**Comment by nfischer** [comment]
No worries! Thanks for reporting this. My favorite bugs are the ones that are already fixed by somewhere else in the code :smile:. And thanks for trying out v0.6!
